llvm/docs/LangRef.rst

```diff
@@ -2994,6 +2994,29 @@ A "convergencectrl" operand bundle is only valid on a ``convergent`` operation.
 When present, the operand bundle must contain exactly one value of token type.
 See the :doc:`ConvergentOperations` document for details.
 
+.. _ob_fpe:
+
+Floating-point Environment Operand Bundles
```

Comment:
Should we place restrictions on which intrinsics these operand bundles can be attached to? It seems that they would have no meaning for non-fp intrinsics, and it's unclear whether they should require the "strictfp" attribute.

Reply:
I think we could introduce additional "memory" resources in addition to inaccessible memory to represent FP environment. In this case intrinsisc like `sqrt` would have effect like "reads control modes" and "sets status bits", which would characterize FP intrinsics. In the case of default FP environment these properties should be erased however.

Hopefully more fine-grained FP environment access could avoid unneeded ordering and increase performance of non-default execution modes.